### PR TITLE
Feature/issue filter

### PR DESCRIPTION
**Pull request by Oksana098**

### CODE REVIEW

File: backend/views/views.py [hunk 1]
```diff
@@ -49,12 +51,14 @@ def user_page():
         search_by = int(request.args.get('search_by'))
         order_by = int(request.args.get('order_by'))
         search_string = str(request.args.get('search'))
+        print 'search_string', search_string
```
Comment (KostSvitlana) [inline backend/views/views.py]: It not need here

File: backend/forms/forms.py [hunk 2]
```diff
@@ -137,3 +137,29 @@ class SearchForm(BaseForm):
...
+    """Search form"""
+
+    search = StringField(
+        'search'
+    )
+    search_by = SelectField(
+        'search_by',
+        choices=[
+            ('name', 'name'),
+            ('category', 'category'),
```
Comment (KostSvitlana) [inline backend/forms/forms.py]: Needless comma

File: backend/forms/forms.py [hunk 3]
```diff
@@ -137,3 +137,29 @@ class SearchForm(BaseForm):
...
+        ]
+    )
+    order_by = SelectField(
+        'order_by',
+        choices=[
+            ('0', 'name'),
+            ('1', 'category'),
```
Comment (KostSvitlana) [inline backend/forms/forms.py:159]: And here

File: backend/templates/issues_page.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+{% block title_suffix %}Issues managements{% endblock title_suffix %}
+
+{% block navbar_li %}
+    <li><a href="{{ url_for('admin') }}">Admin page</a></li>
+    <li><a href="{{ url_for('logout') }}">Logout</a></li>
+{% endblock navbar_li %}
+
+{% block content %}
+    <form action="/admin/issuespage_filter" method="GET">
```
Comment (KostSvitlana) [inline backend/templates/issues_page.html]: Need write {{ url_for('issuespage_filter')}} without 'admin'

File: backend/views/views.py [hunk 5]
```diff
@@ -180,3 +185,106 @@ def logout():
...
+            Issue.user_id == User.id, Issue.category_id == Category.id,
+            Issue.id == count_att.c.id, Issue.id == status.c.id)).order_by(
+                Issue.id).all()
+    return render_template('issues_page.html', issues=issues, form=form)
+
+
+@app.route('/issuespage_filter', methods=['GET', 'POST'])
+@admin_permissions
+def issues_page_filter():
+    """Issues page route."""
```
Comment (KostSvitlana) [inline backend/views/views.py]: Not change docstring

File: backend/views/views.py [hunk 6]
```diff
@@ -180,3 +185,106 @@ def logout():
...
+    if form.validate():
+        search_by = request.args.get('search_by')
+        order_by = int(request.args.get('order_by'))
+        search_string = str(request.args.get('search'))
+
+        if len(search_string) >= MIN_SEARCH_STR:
+            search_parameter = '%{}%'.format(search_string)
+
+            if 'name' == search_by:
+                condition = Issue.name.like(search_parameter)
```
Comment (KostSvitlana) [inline backend/views/views.py]: Search looking for all string. If user type few words with space, search not return results.

File: backend/views/views.py [hunk 7]
```diff
@@ -180,3 +185,106 @@ def logout():
...
+    status = db.session.query(Issue.id, Status.status).filter(and_(
+        IssueHistory.issue_id == Issue.id,
+        IssueHistory.status_id == Status.id, Issue.id == last_date.c.id,
+        IssueHistory.transaction_date == last_date.c.date)).subquery('status')
+
+    issues = db.session.query(
+        Category.category, Issue, User.alias, count_att.c.count,
+        status.c.status).filter(and_(
+            Issue.user_id == User.id, Issue.category_id == Category.id,
+            Issue.id == count_att.c.id, Issue.id == status.c.id, condition)).order_by(order).all()
```
Comment (KostSvitlana) [inline backend/views/views.py]: It is repeat code

File: backend/views/views.py (hunk 6 above)
Comment (Oksana098) [reply backend/views/views.py]: Did you check it? are you sure?
If run project you can that it is working)

File: backend/views/views.py (hunk 1 above)
Comment (Oksana098) [reply backend/views/views.py]: yes

File: backend/templates/issues_page.html (hunk 4 above)
Comment (Oksana098) [reply backend/templates/issues_page.html]: fixed

File: backend/forms/forms.py (hunk 3 above)
Comment (Oksana098) [reply backend/forms/forms.py:159]: yes but it normal

File: backend/forms/forms.py (hunk 2 above)
Comment (Oksana098) [reply backend/forms/forms.py]: yes but it normal

File: backend/views/views.py (hunk 7 above)
Comment (Oksana098) [reply backend/views/views.py]: I integrated my code in your

**Comment by Oksana098** [comment]
I done final changes please check

File: backend/views/views.py [hunk 8]
```diff
@@ -5,6 +5,8 @@
 from sqlalchemy import and_, func, or_
 
 from backend.app import app, db
+from backend.forms.forms import (LoginForm, SearchForm,
```
Comment (KostSvitlana) [inline backend/views/views.py]: I rename my form `SearchForm` to `SearchUserForm`

File: backend/views/views.py [hunk 9]
```diff
@@ -182,10 +183,33 @@ def logout():
...
+        search_by = int(request.args.get('search_by'))
+        order_by = int(request.args.get('order_by'))
+        search_string = str(request.args.get('search'))
+
+        search_list = ['name', 'category']
+
+        if len(search_string) >= MIN_SEARCH_STR:
+            search_parameter = '%{}%'.format(search_string)
+            if 'name' == search_list[search_by]:
+                condition = Issue.name.like(search_parameter)
```
Comment (KostSvitlana) [inline backend/views/views.py]: I recomend function `ilike`. It makes not  case sensitive search

File: backend/views/views.py [hunk 10]
```diff
@@ -182,10 +183,33 @@ def logout():
...
+
+            else:
+                condition = Category.category.like(search_parameter)
+
+        order_list = [Issue.name, Category.category]
+        order = order_list[order_by]
```
Comment (KostSvitlana) [inline backend/views/views.py]: You are not use variables `order` and `condition`. Need write them in sql query

File: backend/views/views.py [hunk 11]
```diff
@@ -182,10 +183,33 @@ def logout():
...
+
+    form = SearchIssuesForm(request.args, meta={'csrf': False})
```
Comment (KostSvitlana) [inline backend/views/views.py]: why are you create second form?

**Review by KostSvitlana — COMMENTED** [review]
Search does not work, you are not transfer form on template

File: backend/views/views.py [hunk 12]
```diff
@@ -5,6 +5,8 @@
 from sqlalchemy import and_, func, or_
 
 from backend.app import app, db
+from backend.forms.forms import (LoginForm, SearchForm,
+                                 UserForm, SearchIssuesForm)
 from backend.forms.forms import LoginForm, SearchUserForm, UserForm
```
Comment (KostSvitlana) [inline backend/views/views.py]: There is repeat of the import

File: backend/views/views.py [hunk 13]
```diff
@@ -182,18 +183,49 @@ def logout():
...
+        issues = db.session.query(
+            Category.category, Issue, User.alias, count_att.c.count).filter(and_(
+                Issue.user_id == User.id, Issue.category_id == Category.id,
+                Issue.id == count_att.c.id, condition)).order_by(order).all()
+
+    else:
+        issues = db.session.query(
+            Category.category, Issue, User.alias, count_att.c.count).filter(and_(
+                Issue.user_id == User.id, Issue.category_id == Category.id,
+                Issue.id == count_att.c.id)).order_by(Issue.name).all()
```
Comment (KostSvitlana) [inline backend/views/views.py:226]: Why `order_by(Issue.name)`? Mayby `Issue.id`?

Comment (Oksana098) [reply backend/views/views.py:226]: I think better filter sorting by name default not by id.
As third parameter of sort we can in future sort by id.

**Review by KostSvitlana — APPROVED** [review]

**Pull request merged**